File: WEEK9/main.cpp
#include <iostream>
#include <string>
#include <algorithm>

using namespace std;



int* intersect(int* arr1, int* arr2, int size1, int size2, int& resultSize) {
    resultSize = size1 < size2 ? size1 : size2;
    int* result = new int[resultSize];
    int k = 0;
    for (int i = 0; i < size1; i++) {
        for (int j = 0; j < size2; j++) {
            if (arr1[i] == arr2[j]) {
                result[k++] = arr1[i];
                break;
            }
        }
    }
    return result;
}

bool isPalindrome(string str) {
    int left = 0, right = str.length() - 1;
    while (left < right) {
        if (str[left] != str[right]) {
            return false;
        }
        left++;
        right--;
    }
    return true;
}

string* getPalindromes(string* arr, int size, int& palindromesCount) {
    palindromesCount = 0;
    for(int i =0;i<size;i++){
        if(isPalindrome(arr[i])){
            palindromesCount++;
        }
    }
    string* newArray= new string[palindromesCount];
    for(int i = 0; i<palindromesCount;i++){
        if(isPalindrome(arr[i])){
            newArray[i]=arr[i];
        }
    }

    return newArray;
}

int* SieveOfEratosthenes(int* arr, int n) {
    int* primes = new int[n];
    int count = 0;
    for (int i = 0; i < n; i++) {
        if (arr[i] == 0) {  // skip composite numbers
            continue;
        }
        primes[count++] = arr[i];  // add prime number to array
        for (int j = i+1; j < n; j++) {
            if (arr[j] % arr[i] == 0) {  // mark multiples of prime as composite
                arr[j] = 0;
            }
        }
    }
    int* result = new int[count];
    for (int i = 0; i < count; i++) {
        result[i] = primes[i];
    }
    delete[] primes;
    return result;
}

string* anagrams(string word, string* word_list, int sizeOfWordList, int& numberForAnagrams) {
    
    string sorted_word = word;
    sort(sorted_word.begin(), sorted_word.end());
    
    
    string* anagrams = new string[sizeOfWordList];
    numberForAnagrams = 0;
    for (int i = 0; i < sizeOfWordList; i++) {
        
        string sorted_w = word_list[i];
        sort(sorted_w.begin(), sorted_w.end());
        
        
        if (sorted_w == sorted_word) {
            anagrams[numberForAnagrams] = word_list[i];
            numberForAnagrams++;
        }
    }
    
    return anagrams;
}

















int main() {

    
    
    return 0;
}

File: WEEK9/main.py
import random

def function1(list1,list2):
    list3=[]
    for element in list1:
        if element in list2:
            list3.append(element)
    return list3



def isPalindrome(word):
    i=0
    for i in range(len(word)//2):
        if word[i] == word[-1 - i]:
            i += 1
        else:
            return False
    return True


def function2(list1):
    newList=[]
    i=0
    for word in list1:
        if isPalindrome(word):
            newList.append(word)


def  function3(list1):
    list1=sorted(list1)
    markedNumbers=[]
    primeNumbers=[]
    for number in list1:

        if number not in markedNumbers:
            if number >1:
                primeNumbers.append(number)
                for list1Number in list1:
                    if not list1Number in primeNumbers:
                        if not list1Number in markedNumbers:
                            for primeNumber in primeNumbers:
                                if list1Number%primeNumber==0:
                                    markedNumbers.append(list1Number)

    return primeNumbers


numbers=[]

for i in range(100):
    numbers.append(random.randint(0,100))

numbers=function3(numbers)
print(numbers)


def anagrams(word, word_list):

    sorted_word = sorted(word)

    anagrams = []

    for wordOflist in word_list:

        sorted_w = sorted(wordOflist)

        if sorted_word == sorted_w:

            anagrams.append(wordOflist)

    return anagrams







